observer/probers/crl/crl_conf.go

```diff
@@ -0,0 +1,132 @@
+func (c CRLConf) UnmarshalSettings(settings []byte) (probers.Configurer, error) {
+	var conf CRLConf
+	err := yaml.Unmarshal(settings, &conf)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (c CRLConf) validateURL() error {
+	url, err := url.Parse(c.URL)
+	if err != nil {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, expected a valid url", c.URL)
+	}
+	if url.Scheme == "" {
+		return fmt.Errorf(
+			"invalid 'url', got: %q, missing scheme", c.URL)
+	}
+	return nil
+}
+
+// MakeProber constructs a `CRLProbe` object from the contents of the
+// bound `CRLConf` object. If the `CRLConf` cannot be validated, an
+// error appropriate for end-user consumption is returned instead.
+func (c CRLConf) MakeProber(collectors map[string]prometheus.Collector) (probers.Prober, error) { // validate `url` err := c.validateURL()
+	// validate `url`
+	err := c.validateURL()
+	if err != nil {
+		return nil, err
+	}
+	probe := CRLProbe{c.URL, nil, nil, nil}
+
+	// validate the prometheus collectors that were passed in
+	var ok bool
+	for name, coll := range collectors {
+		switch name {
+		case nextUpdateName:
+			probe.cNextUpdate, ok = coll.(*prometheus.GaugeVec)
+			if !ok {
+				return nil, fmt.Errorf("crl prober received collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, coll)
+			}
+		case thisUpdateName:
+			probe.cThisUpdate, ok = coll.(*prometheus.GaugeVec)
+			if !ok {
+				return nil, fmt.Errorf("crl prober received collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, coll)
+			}
+		case certCountName:
+			probe.cCertCount, ok = coll.(*prometheus.GaugeVec)
+			if !ok {
+				return nil, fmt.Errorf("crl prober received collector '%s' of wrong type, got: %T, expected *prometheus.GaugeVec", name, coll)
+			}
+		default:
+			return nil, fmt.Errorf("crl prober received unexpected collector '%s'", name)
+		}
+	}
+	if probe.cNextUpdate == nil {
+		return nil, fmt.Errorf("crl prober did not receive collector '%s'", nextUpdateName)
+	}
+	if probe.cThisUpdate == nil {
+		return nil, fmt.Errorf("crl prober did not receive collector '%s'", thisUpdateName)
+	}
+	if probe.cCertCount == nil {
+		return nil, fmt.Errorf("crl prober did not receive collector '%s'", certCountName)
+	}
+
+	return probe, nil
+}
+
+// Instrument constructs any `prometheus.Collector` objects the `CRLProbe` will
+// need to report its own metrics. A map is returned containing the constructed
+// objects, indexed by the name of the prometheus metric. If no objects were
+// constructed, nil is returned.
+func (c CRLConf) Instrument() map[string]prometheus.Collector {
+	nextUpdate := prometheus.Collector(prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: nextUpdateName,
+			Help: "CRL nextUpdate unix timestamp",
+		}, []string{"url"},
+	))
+	thisUpdate := prometheus.Collector(prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: thisUpdateName,
+			Help: "CRL thisUpdate unix timestamp",
```

Comment:
```suggestion
			Help: "CRL thisUpdate Unix timestamp in seconds (int64)",
```

Reply:
Re: an offline convo, go ahead and omit the int64 bit here since all prometheus values will be float64. I mostly just wanted folks seeing this Help to know that it's seconds but you'll need to do some conversion (timestamp * 1000) since Grafana expects Unix timestamps in milliseconds, not seconds